project3/coordinator/messenger.cpp

```diff
@@ -5,71 +5,62 @@
 #include "messenger.h"
 
 #include <chrono>
-#include <utility>
-
 #include <loguru.hpp>
+#include <utility>
 
 namespace coordinator {
 
-Messenger::Messenger(
-    std::shared_ptr<queue::Queue<MessageLog::Message>> msg_queue,
-    std::shared_ptr<MessageLog> msg_log,
-    std::shared_ptr<ConnectedParticipants::Participant> participant,
-    std::shared_ptr<ConnectedParticipants> connected_participants)
-    : msg_queue_(std::move(msg_queue)),
-      msg_log_(std::move(msg_log)),
-      connected_participants_(std::move(connected_participants)),
-      participant_(std::move(participant)) {}
+Messenger::Messenger(std::shared_ptr<MessageLog> msg_log,
+                     ConnectedParticipants::Participant participant)
+    : msg_log_(std::move(msg_log)), participant_(std::move(participant)) {}
+bool Messenger::operator==(const Messenger& messenger) const {
```

Comment:
If you're using `shared_ptr`s in an `unordered_set`, you probably don't need this overload...